step-templates/octopus-serialize-space-to-terraform.json

```diff
@@ -0,0 +1,125 @@
for more information.\")\n    sys.exit(1)\n\ndate = datetime.now().strftime('%Y.%m.%d.%H%M%S')\n\nprint('Looking up space name')\nurl = parser.server_url + '/api/Spaces/' + parser.space_id\nheaders = {\n    'X-Octopus-ApiKey': parser.api_key,\n    'Accept': 'application/json'\n}\nrequest = urllib.request.Request(url, headers=headers)\n\n# Retry the request for up to a minute.\nresponse = None\nfor x in range(12):\n    response = urllib.request.urlopen(request)\n    if response.getcode() == 200:\n        break\n    time.sleep(5)\n\nif not response or not response.getcode() == 200:\n    print('The API query failed')\n    sys.exit(1)\n\ndata = json.loads(response.read().decode(\"utf-8\"))\nprint('Space name is ' + data['Name'])\n\nprint(\"Creating Terraform module package\")\nif is_windows():\n    execute(['octo',\n             'pack',\n             '--format', 'zip',\n             '--id', re.sub('[^0-9a-zA-Z]', '_', data['Name']),\n             '--version', date,\n             '--basePath', os.getcwd() + '\\\\export',\n             '--outFolder', 'C:\\\\export'])\nelse:\n    _, _, _ = execute(['docker', 'run',\n                            '--rm',\n                            '--add-host=' + parsed_url.hostname + ':' + octopus.strip(),\n                            '-v', os.getcwd() + \"/export:/export\",\n                            'octopusdeploy/octo',\n                            'pack',\n                            '--format', 'zip',\n                            '--id', re.sub('[^0-9a-zA-Z]', '_', data['Name']),\n                            '--version', date,\n                            '--basePath', '/export',\n                            '--outFolder', '/export'])\n\nprint(\"Uploading Terraform module package\")\nif is_windows():\n    _, _, _ = execute(['octo',\n                            'push',\n                            '--apiKey', parser.api_key,\n                            '--server', parser.server_url,\n                            '--space', parser.upload_space_id,\n                            '--package', 'C:\\\\export\\\\' +\n                            re.sub('[^0-9a-zA-Z]', '_', data['Name']) + '.' + date + '.zip',\n                            '--replace-existing'])\nelse:\n    _, _, _ = execute(['docker', 'run',\n                            '--rm',\n                            '--add-host=' + parsed_url.hostname + ':' + octopus.strip(),\n                            '-v', os.getcwd() + \"/export:/export\",\n                            'octopusdeploy/octo',\n                            'push',\n                            '--apiKey', parser.api_key,\n                            '--server', parser.server_url,\n                            '--space', parser.upload_space_id,\n                            '--package', '/export/' +\n                            re.sub('[^0-9a-zA-Z]', '_', data['Name']) + '.' + date + '.zip',\n                            '--replace-existing'])\n\nprint(\"##octopus[stdout-default]\")\n\nprint(\"Done\")\n",
```

Comment:
Minor query: Should the second instances of the help printed here refer to `SerializeSpace.` (instead of `ThisInstance.` or `SerializeProject.`) ?

```py
if len(parser.server_url) == 0:
    print("--server-url, ThisInstance.Server.Url, or SerializeProject.ThisInstance.Server.Url must be defined")
    sys.exit(1)

if len(parser.api_key) == 0:
    print("--api-key, ThisInstance.Api.Key, or ThisInstance.Api.Key must be defined")
    sys.exit(1)
```

Reply:
:+1: They should, this was a copy/paste oversight. I'll fix that up.